controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -128,12 +131,50 @@ public void initializeStreamWriters(final ClientFactory clientFactory,
      * @param contextOpt optional context
      * @return update status.
      */
-    @Task(name = "updateConfig", version = "1.0", resource = "{scope}/{stream}")
     public CompletableFuture<UpdateStreamStatus.Status> updateStream(String scope, String stream, StreamConfiguration config, OperationContext contextOpt) {
-        return execute(
-                new Resource(scope, stream),
-                new Serializable[]{scope, stream, config, null},
-                () -> updateStreamConfigBody(scope, stream, config, contextOpt));
+        AtomicReference<UpdateStreamEvent> eventRef = new AtomicReference<>(null);
+        // 1. get configuration
```

Comment:
Some private methods would help here.
IE: each of these 1, 2,3,4 could be a method, and the name would implicitly convey the step.

Reply:
More logging, please. Please also consider adding more metrics.

Reply:
done refactoring.
Not added metrics though.